python/pip_install/tools/wheel_installer/wheel.py

```diff
@@ -583,6 +583,17 @@ def version(self) -> str:
         # TODO Also available as installer.sources.WheelSource.version
         return str(self.metadata["Version"])
 
+    @property
+    def record(self) -> dict[str, tuple[str, int]]:
+        with installer.sources.WheelFile.open(self.path) as wheel_source:
+            record_content = wheel_source.read_dist_info("RECORD")
+        return {
+            file: (filehash, int(filelen))
+            for line in record_content.splitlines()
+            for file, filehash, filelen in [line.split(",")]
+            if filehash  # Skip RECORD itself, which has no hash or length
+        }
```

Comment:
Just having a tool that uses the installer directly instead of reusing this class would be preferable due to reasons mentioned elsewhere.

Reply:
moved this to the tool itself. Don't actually need the installer dep as well, I just roughly replicated https://github.com/pypa/installer/blob/501bdc2ae800a27e362186fe72928f8b5c206e00/src/installer/sources.py#L179-L193